### PR TITLE
Undo PokeCodec's PRs and port changes

### PR DESCRIPTION
**Pull request by aaaaaa123456789**
Undo most of PokeCodec's PRs after the debate in chat. Some harmless or completely superseded PRs have been left alone, as there is not much benefit in attempting to undo them.

Thanks to @Kurausukun for porting back matched functions from those PRs.

Reverts #1104, #1108, #1115, #1118, #1119, #1124, #1126, #1127, #1132, #1136, #1137, #1139, #1140, #1144, #1148, #1149, #1150, #1153, #1155, #1177, #1179, #1180, #1181, #1182 and #1183.

Matched functions (from list below):

* [x] _InitContestMonPixels
* [x] AnimFallingFeather
* [x] AnimTask_CreateSurfWave
* [x] AnimTask_CreateSurfWave_Step1
* [x] BattleIntroOpponent1SendsOutMonAnimation
* [x] BerryCrushCommand_BeginNormalPaletteFade
* [x] CheckBagHasSpace
* [x] CheckFocusPunch_ClearVarsBeforeTurnStarts
* [x] Cmd_forcerandomswitch
* [x] Cmd_unknown_24
* [x] CopyGlyphToWindow
* [x] CreateFrontierBrainPokemon
* [x] CreateWaterPulseRingBubbles
* [x] DrawResultsTextWindow
* [x] GetConnectedChildStrength
* [x] GetFrontierOpponentClass
* [x] GetTypeEffectivenessPoints
* [x] GLYPH_COPY
* [x] HandleSendFailure
* [x] HasNoMonsToSwitch
* [x] InitObjectEventStateFromTemplate
* [x] InitPaintingMonOamData
* [x] ModifyStatByNature
* [x] MovementAction_StoreAndLockAnim_Step0
* [x] Phase2_FrontierLogoWave_Func4
* [x] PrintMonWeight
* [x] ReceiveDaycareMailData
* [x] RotateAuroraRingColors_Step
* [x] SetSearchRectHighlight
* [x] sub_806F160
* [x] sub_80C6104
* [x] sub_8181C2C
* [x] sub_81BF2B8
* [x] sub_81C79BC
* [x] sub_81D5F58
* [x] TrendyPhraseIsOld
* [x] TryPutLinkBattleTvShowOnAir
* [x] UpdateSaveAddresses

### CODE REVIEW

**Comment by PikalaxALT** [comment]
My approval on this PR, once made ready for review, constitutes a negation of my approvals on any PR mentioned above or below which would be reverted herein.

File: gflib/text.h [hunk 1]
```diff
@@ -269,20 +269,13 @@
...
-};
-
 enum
 {
     FONTATTR_MAX_LETTER_WIDTH,
     FONTATTR_MAX_LETTER_HEIGHT,
     FONTATTR_LETTER_SPACING,
     FONTATTR_LINE_SPACING,
-    FONTATTR_STYLE,
+    FONTATTR_UNKNOWN,   // dunno what this is yet
```
Comment (Poliwrath) [inline gflib/text.h:278]: Why undo this change? Is it still unknown?

Comment (aaaaaa123456789) [reply gflib/text.h:278]: Because it's a 10k PR and we need to limit scope if it's going to ever happen.

Comment (Poliwrath) [reply gflib/text.h:278]: It was added in #1149, I don't see any reason to undo that...? (unless it's completely wrong)

Comment (GriffinRichards) [reply gflib/text.h:278]: This is a draft PR started by reverting the listed PRs in full, please refrain from commenting on specific reversions until it's complete

Comment (Poliwrath) [reply gflib/text.h:278]: I'm mostly asking why are these seemingly okay changes being reverted, seems pretty counterintuitive to replace named fields with "UNKNOWN"

Comment (GriffinRichards) [reply gflib/text.h:278]: As I said, the PR will be pared down to useful changes. The PR is a draft, it's incomplete, and the field in question is never read regardless, so naming it doesn't matter much whether it's kept or not

Comment (Poliwrath) [reply gflib/text.h:278]: Ohhh, gotcha.

**Comment by Kurausukun** [comment]
Preliminary list of functions that have had their matching status affected by this (may include false positives):
```
RotateAuroraRingColors_Step
InitObjectEventStateFromTemplate
CreateWaterPulseRingBubbles
AnimTask_AnimateGustTornadoPalette_Step
DestroyAnimSpriteAfterTimer
AnimFallingFeather
sub_810EB40
AnimTask_DrillPeckHitSplats
AnimDiveWaterSplash
AnimSprayWaterDroplet
AnimSkyAttackBird
unref_sub_810F184
GLYPH_COPY
CopyGlyphToWindow
AnimTask_CreateSurfWave
AnimTask_CreateSurfWave_Step1
GetConnectedChildStrength
Cmd_unknown_24
rfu_LMAN_linkWatcher
sub_81D5F58
_InitContestMonPixels
BattleLoadSubstituteOrMonSpriteGfx
SetActionsAndBattlersTurnOrder
CheckFocusPunch_ClearVarsBeforeTurnStarts
RunTurnActionsFunctions
HandleSendFailure
GetConditionMenuMonString
GetTypeEffectivenessPoints
BattleTv_SetDataBasedOnAnimation
TryPutLinkBattleTvShowOnAir
DrawResultsTextWindow
CreateFrontierBrainPokemon
sub_81BF2B8
SetSearchRectHighlight
sub_81C79BC
UpdateSaveAddresses
MovementAction_StoreAndLockAnim_Step0
CheckBagHasSpace
ModifyStatByNature
BattleIntroOpponent1SendsOutMonAnimation
Phase2_FrontierLogoWave_Func4
HasNoMonsToSwitch
LoadTrainerHillFloorObjectEventScripts
Cmd_forcerandomswitch
GetFrontierOpponentClass
sub_806F160
sub_8181C2C
ReceiveDaycareMailData
BerryCrushCommand_BeginNormalPaletteFade
InitPaintingMonOamData
TrendyPhraseIsOld
sub_80C6104
PrintMonWeight
```

**Comment by aaaaaa123456789** [comment]
Checked off all functions up to this point.

**Comment by aaaaaa123456789** [comment]
That should be all of the functions; I'll try to resolve merge conflicts soon.

**Comment by aaaaaa123456789** [comment]
Ready for review after CI is done. Thanks to @Kurausukun for the effort in porting back the matching changes.

**Comment by PokeCodec** [comment]
CI failed. Try again?

**Comment by aaaaaa123456789** [comment]
Failed due to dkP errors. The solution would be to use a good toolchain, but that's what we have...

File: src/contest.c [hunk 2]
```diff
@@ -998,6 +998,8 @@ void ResetLinkContestBoolean(void)
 
 static void SetupContestGpuRegs(void)
 {
+    u16 savedIme;
```
Comment (PokeCodec) [inline src/contest.c:1001]: This variable is not used at all and the build matches with this gone. Since the last travis build failed anyway, why not add this
```suggestion
```

**Review by PokeCodec — COMMENTED** [review]
This is my only request really

File: src/digit_obj_util.c [hunk 3]
```diff
@@ -33,7 +33,7 @@ struct DigitPrinterAlloc
 };
 
 // this file's functions
-static u8 GetFirstOamId(u8 oamCount);
+static u8 GetFirstOamId(u8 oamCount);;
```
Comment (PokeCodec) [inline src/digit_obj_util.c:36]:
```suggestion
static u8 GetFirstOamId(u8 oamCount);
```

Comment (PokeCodec) [reply src/digit_obj_util.c:36]: Extra semicolon added

File: src/contest.c [hunk 4]
```diff
@@ -1492,7 +1492,7 @@ static void Task_DisplayAppealNumberText(u8 taskId)
 static void Task_TryShowMoveSelectScreen(u8 taskId)
 {
     // Wait for button press to show move select screen
-    if ((JOY_NEW(A_BUTTON)) || (gMain.newKeys == B_BUTTON))
+    if ((gMain.newKeys & A_BUTTON) || (gMain.newKeys == B_BUTTON))
```
Comment (PikalaxALT) [inline src/contest.c:1495]: TODO: Reintroduce the JOY_XXX macro changes

**Review by PikalaxALT — APPROVED** [review]

File: src/battle_anim_dark.c [hunk 5]
```diff
@@ -635,6 +635,8 @@ void AnimTask_MoveTargetMementoShadow(u8 taskId)
 
 static void AnimTask_MoveTargetMementoShadow_Step(u8 taskId)
 {
+    u8 pos;
+    u16 i;
```
Comment (GriffinRichards) [inline src/battle_anim_dark.c:639]: Unnecessary

File: src/battle_anim_bug.c [hunk 6]
```diff
@@ -372,20 +372,28 @@ static void AnimTranslateStinger(struct Sprite *sprite)
...
         {
-            gBattleAnimArgs[2] *= -1;
-            gBattleAnimArgs[0] *= -1;
+            s16 temp1, temp2;
+
+            temp1 = gBattleAnimArgs[2];
+            gBattleAnimArgs[2] = -temp1;
+
+            temp2 = gBattleAnimArgs[0];
+            gBattleAnimArgs[0] = -temp2;
```
Comment (GriffinRichards) [inline src/battle_anim_bug.c:396]: I would consider this a fakematch resolution that should be kept

File: src/battle_anim_effects_3.c [hunk 7]
```diff
@@ -5002,11 +5002,11 @@ void AnimTask_GetReturnPowerLevel(u8 taskId)
...
 {
     u8 spriteId, spriteId2;
-    u32 personality;
-    u32 otId;
+    int personality;
+    int otId;
     u16 species;
     u8 subpriority;
-    bool8 isBackPic;
+    u8 isBackPic;
```
Comment (GriffinRichards) [inline src/battle_anim_effects_3.c:5009]: Keep `bool8`, and its TRUE/FALSE assignments

File: include/battle_main.h [hunk 8]
```diff
@@ -73,6 +73,8 @@ void nullsub_20(void);
 void BeginBattleIntro(void);
 void SwitchInClearSetData(void);
 void FaintClearSetData(void);
+void sub_803B3AC(void); // unused
+void sub_803B598(void); // unused
```
Comment (GriffinRichards) [inline include/battle_main.h:77]: No need to put these back

File: src/battle_anim_effects_3.c [hunk 9]
```diff
@@ -5134,7 +5137,7 @@ void AnimTask_SnatchOpposingMonMove(u8 taskId)
                 gSprites[spriteId].pos2.x = 0;
         }
 
-        gTasks[taskId].data[1] &= 0xFF;
+        gTasks[taskId].data[1] = (u8)gTasks[taskId].data[1];
```
Comment (GriffinRichards) [inline src/battle_anim_effects_3.c:5140]: Keep this fakematch resolution

File: src/battle_anim_flying.c [hunk 10]
```diff
@@ -1180,7 +1199,7 @@ static void AnimSkyAttackBird(struct Sprite *sprite)
     sprite->data[7] = ((posy - sprite->pos1.y) << 4) / 12;
 
     rotation = ArcTan2Neg(posx - sprite->pos1.x, posy - sprite->pos1.y);
-    rotation -= 16384;
+    rotation += 49152;
```
Comment (GriffinRichards) [inline src/battle_anim_flying.c:1202]: Keep

File: src/battle_anim_psychic.c [hunk 11]
```diff
@@ -1101,7 +1101,7 @@ static void AnimTask_TransparentCloneGrowAndShrink_Step(u8 taskId)
         break;
     case 1:
         task->data[1] -= 4;
-        task->data[2] = 256 - (gSineTable[task->data[1]] >> 1);
+        task->data[2] = 256 - (gSineTable[task->data[1]] >> 1);;
```
Comment (GriffinRichards) [inline src/battle_anim_psychic.c:1104]: Keep

File: src/battle_anim_utility_funcs.c [hunk 12]
```diff
@@ -569,7 +569,7 @@ static void sub_81172EC(u8 taskId)
             if (gTasks[taskId].data[12] == 0)
             {
                 sub_80A477C(0);
-                gTasks[taskId].data[15]++;
+                gTasks[taskId].data[15]++;;
```
Comment (GriffinRichards) [inline src/battle_anim_utility_funcs.c:572]: Keep

File: src/battle_records.c [hunk 13]
```diff
@@ -485,7 +485,7 @@ static void CB2_ShowTrainerHillRecords(void)
         gMain.state++;
         break;
     case 2:
-        sTilemapBuffer = AllocZeroed(BG_SCREEN_SIZE);
+        sTilemapBuffer = AllocZeroed(0x800);
```
Comment (GriffinRichards) [inline src/battle_records.c:488]: Keep all the instances of this

File: src/contest.c [hunk 14]
```diff
@@ -4136,12 +4139,12 @@ static u8 CreateContestantBoxBlinkSprites(u8 contestant)
     CpuFill32(0, gContestResources->boxBlinkTiles2 + 0x500, 0x300);
 
     RequestDma3Copy(gContestResources->boxBlinkTiles1,
-                    (u8 *)(OBJ_VRAM0 + gSprites[spriteId1].oam.tileNum * 32),
+                    (u8 *)(VRAM + 0x10000 + gSprites[spriteId1].oam.tileNum * 32),
```
Comment (GriffinRichards) [inline src/contest.c:4142]: Keep all the instances of this

File: src/event_object_movement.c [hunk 15]
```diff
@@ -8507,11 +8509,23 @@ void sub_809783C(struct Sprite *sprite, u8 a2, u8 a3, u8 a4)
...
-    s16 v5[] = {16, 16, 32};
-    u8 v6[] = {0, 0, 1};
-    u8 v2 = 0;
+    s16 v5[3];
+    u8 v6[3];
+    u8 v2;
+
+    memcpy(v5, gUnknown_0850E840, 6); // TODO: get rid of memcpy
+    memcpy(v6, gUnknown_0850E846, 3);
+    v2 = 0;
```
Comment (GriffinRichards) [inline src/event_object_movement.c:8528]: Keep this fakematch resolution

File: src/event_object_movement.c [hunk 16]
```diff
@@ -8532,11 +8546,23 @@ u8 sub_809785C(struct Sprite *sprite)
...
-    s16 v5[] = {32, 32, 64};
-    u8 v6[] = {1, 1, 2};
-    u8 v2 = 0;
+    s16 v5[3];
+    u8 v6[3];
+    u8 v2;
+
+    memcpy(v5, gUnknown_0850E84A, 6);
+    memcpy(v6, gUnknown_0850E850, 3);
+    v2 = 0;
```
Comment (GriffinRichards) [inline src/event_object_movement.c:8565]: Same here

File: src/m4a.c [hunk 17]
```diff
@@ -598,21 +614,28 @@ void MPlayStart(struct MusicPlayerInfo *mplayInfo, struct SongHeader *songHeader
...
+        while (i < mplayInfo->trackCount)
         {
             TrackStop(mplayInfo, track);
             track->flags = 0;
+            i++;
+            track++;
         }
 
-        if (songHeader->reverb & SOUND_MODE_REVERB_SET)
+        if (songHeader->reverb & 0x80)
```
Comment (GriffinRichards) [inline src/m4a.c:638]: Keep

File: src/overworld.c [hunk 18]
```diff
@@ -3187,7 +3188,7 @@ static void SpriteCB_LinkPlayer(struct Sprite *sprite)
     SetObjectSubpriorityByZCoord(objEvent->previousElevation, sprite, 1);
     sprite->oam.priority = ZCoordToPriority(objEvent->previousElevation);
 
-    if (linkPlayerObjEvent->movementMode == MOVEMENT_MODE_FREE)
+    if (!linkPlayerObjEvent->movementMode != MOVEMENT_MODE_FREE)
```
Comment (GriffinRichards) [inline src/overworld.c:3191]: Keep

File: src/pokenav_main_menu.c [hunk 19]
```diff
@@ -436,10 +436,9 @@ static u32 LoopedTask_SlideMenuHeaderDown(s32 a0)
     if (ChangeBgY(0, 384, 2) <= 0)
     {
         ChangeBgY(0, 0, 0);
-        return LT_FINISH;
+        return 4;
     }
-
-    return LT_PAUSE;
+    return 2;
```
Comment (GriffinRichards) [inline src/pokenav_main_menu.c:441]: Keep

File: src/save_failed_screen.c [hunk 20]
```diff
@@ -349,7 +349,7 @@ static void VBlankCB_UpdateClockGraphics(void)
 
     gMain.oamBuffer[0] = sClockOamData;
     gMain.oamBuffer[0].x = 112;
-    gMain.oamBuffer[0].y = (CLOCK_WIN_TOP + 1) * 8;
+    gMain.oamBuffer[0].y = (CLOCK_WIN_TOP + 1) * 8;;
```
Comment (GriffinRichards) [inline src/save_failed_screen.c:352]: Keep

**Review by GriffinRichards — CHANGES_REQUESTED** [review]
More of these reversions should be pared down, mostly constant usage and fakematch fixes.

File: src/contest.c (hunk 4 above)
Comment (GriffinRichards) [reply src/contest.c:1495]: This should be done in this PR

Comment (aaaaaa123456789) [reply src/contest.c]: Done

**Review by GriffinRichards — APPROVED** [review]
